### DISCUSSION TITLE
Validate custom public claims

### DISCUSSION THREAD

**Discussion by ghost** [panva/jose#238] [Ideas]
Hi panva, how r u last time?

I'm using the jose lib to validate JWT tokens emmited from the oidc-provider. I'm using a stateless validation. The problem for me, is that jose aren't validating custom public claims on payload.

This claim are part of specs and RFCs, but, something make me feel that there are some reason because you don't accepts custom claims to validate playload.

There are a lot of others libs that make the job, but it's really important keep the stack a save! I prefer to use your libs.

Thanks a lot!

Regards.

**Comment by panva** [comment]
Please make a concrete, detailed suggestion.

**Reply by ghost** [reply]
Iam suggesting this:

  const options = {
    issuer: issuers,
    subject,
   'urn:example:customclaim': 'XXX'
  };

  const jwtToken = await jwtVerify(accessToken, JWKS, options);

**Reply by panva** [reply]
Wouldn't that the same as

```js
const jwtToken = await jwtVerify(accessToken, JWKS, { issuer, subject }});

assert.equal(jwtToken.payload['urn:example:customclaim'], 'xxx')
```

Custom claims have no defined format or logic to be validated, rather than building a "matcher"/"verifier" system I think this should remain in your application's code to validate.

**Reply by ghost** [reply]
Potable, that sounds good.

So, by the way, thanks for fastest responses.

Regards

**Reply by mbrevda** [reply]
was looking for this, too. Use case: we accept tokens from multiple providers, and end up in a loop catching `JWKSMultipleMatchingKeys` . Being able to define the verifier claims and handle the logic in a single location would be helpful.
